### PR TITLE
Fixed #24145 -- Changed POST redirect error to be raised on PUT and PATCH too

### PR DESCRIPTION
**Pull request by samuelcolvin**
ref #24145

### CODE REVIEW

**Comment by samuelcolvin** [comment]
ref tomchristie/django-rest-framework#2405

**Comment by collinanderson** [comment]
Or how about anything other than GET, HEAD, or OPTIONS?

**Comment by samuelcolvin** [comment]
Can GET, HEAD, or OPTIONS legitimately have data in the request?

**Comment by MarkusH** [comment]
No, only POST, PUT and PATCH can contain data.

Change looks good to me. Could you squash the commits and change the commit message according to our [commit guidelines](https://docs.djangoproject.com/en/dev/internals/contributing/committing-code/#committing-guidelines). Thanks.

File: django/middleware/common.py [hunk 1]
```diff
@@ -75,14 +75,15 @@ def process_request(self, request):
             if (not urlresolvers.is_valid_path(request.path_info, urlconf) and
                     urlresolvers.is_valid_path("%s/" % request.path_info, urlconf)):
                 new_url[1] = new_url[1] + '/'
-                if settings.DEBUG and request.method == 'POST':
+                if settings.DEBUG and request.method in ('POST', 'PUT', 'PATCH'):
+                    url = new_url[0] + new_url[1]
```
Comment (berkerpeksag) [inline django/middleware/common.py]: or `''.join(newurll)`. Also, you could just use it in line 86:

```py
% {'method': request.method, 'url': ''.join(newurll)}
```

Comment (samuelcolvin) [inline django/middleware/common.py]: Fixed as requested.

**Comment by berkerpeksag** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
Shouldn't there be some tests for this?

**Comment by samuelcolvin** [comment]
Perhaps different issue. I definitely don't have time to write tests for this at the moment, so if so it would have to wait.

**Comment by apollo13** [comment]
Tests for a change are certainly not a different issue, the current ones should be extended.

**Comment by timgraham** [comment]
Samuel, do you want to keep the ticket assigned to yourself or should we see if someone else can add a test?

**Comment by samuelcolvin** [comment]
As @apollo13 and I said it makes sense to do the tests in a different issue.

**Comment by apollo13** [comment]
@samuelcolvin I said exactly the opposite of that :þ

**Comment by samuelcolvin** [comment]
Unless extending them would be very easy, where are they?

**Comment by timgraham** [comment]
Existing test: https://github.com/django/django/blob/f60973111806100d284d41c12206c04740063549/tests/middleware/tests.py#L67-L77

**Comment by samuelcolvin** [comment]
tests extended, is this all we need?

**Comment by timgraham** [comment]
Yes, merged in 5b74134f27eabf92870e1c5e81f9e4999f113eab, thanks.

**Pull request closed without merging**